.github/workflows/clients-go-build.yml

```diff
@@ -0,0 +1,136 @@
+name: "clients/go: Build"
+
+defaults:
+  run:
+    working-directory: ./src/clients/go
+
+on:
+  push:
+    branches:
+      - '**'
+    tags:
+      - '*'
+    # paths:
+    #   - src/clients/go/**
+  pull_request:
+    branches:
+      - 'main'
+
+jobs:
+  shellcheck_scripts:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - run: ./scripts/shellcheck.sh
+
+  binaries:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - run: ./tigerbeetle/scripts/install_zig.sh
+      - run: ./scripts/rebuild_binaries.sh
+
+      - name: Cache output
+        uses: actions/cache@v3
+        env:
+          cache-name: cache-go-binaries
+        with:
+          path: src/clients/go/pkg/native
+          key: go-binaries-${{ github.sha }}
+          restore-keys: |
+            go-binaries-${{ github.sha }}
+
+  tests:
+    needs: binaries
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Cache output
+        uses: actions/cache@v3
+        env:
+          cache-name: cache-go-binaries
+        with:
+          path: src/clients/go/pkg/native
+          key: go-binaries-${{ github.sha }}
+          restore-keys: |
+            go-binaries-${{ github.sha }}
+      - run: go test ./...
+
+  # TODO: Windows exit codes broken
```

Comment:
```
Set-StrictMode -Version 3
$ErrorActionPreference = "Stop"
$PSDefaultParameterValues['*:ErrorAction']='Stop'
$LASTEXITCODE = 0
```

Here's what I do [elsewhere](https://github.com/multiprocessio/datastation/blob/main/scripts/ci/prepare_windows.ps1).

Reply:
Cool thanks - I'll get this sorted